wandb/apis/public.py

```diff
@@ -2782,7 +2783,7 @@ def download(self, root=".", replace=False):
             `ValueError` if file already exists and replace=False
         """
         path = os.path.join(root, self.name)
-        if os.path.exists(path) and not replace:
+        if os.path.exists(path) and not replace and not exist_ok:
```

Comment:
I don't think you are handling the case where the file exists and `exist_ok==True`, with your current change you will still re-download it (based on your description, you said you want to avoid downloading if the file exists)

Reply:
@kptkin Good catch. Sorry, was a bit hasty with this PR.

Reply:
No worries :) That's why we have reviews